vernac/vernacentries.ml

```diff
@@ -2291,36 +2291,30 @@ let vernac_timeout ?timeout (f : 'a -> 'b) (x : 'a) : 'b =
     f x
 
 (* Fail *)
-exception HasNotFailed
-exception HasFailed of Pp.t
-
 let test_mode = ref false
 
-(* XXX STATE: this type hints that restoring the state should be the
-   caller's responsibility *)
-let with_fail ~st f =
+(* Restoring the state is the caller's responsibility *)
```

Comment:
This comment confused me considering we immediately shadow with a wrapper to restore state. Maybe it should indicate that this `with_fail` isn't the full implementation of Fail and mention the wrapper.

Reply:
See the first commit, maybe the second commit is not worth it and we should drop it?

I wanted to hint at a "functional" implementation, but indeed, maybe that just makes things more confusing.